internal/v1/handler_blueprints.go

```diff
@@ -191,22 +197,46 @@ func (h *Handlers) ComposeBlueprint(ctx echo.Context, id openapi_types.UUID) err
 	}
 	composeResponses := make([]ComposeResponse, 0, len(blueprint.ImageRequests))
 	clientId := ClientId("api")
-	for _, imageRequest := range blueprint.ImageRequests {
-		composeRequest := ComposeRequest{
-			Customizations:   &blueprint.Customizations,
-			Distribution:     blueprint.Distribution,
-			ImageRequests:    []ImageRequest{imageRequest},
-			ImageName:        &blueprintEntry.Name,
-			ImageDescription: &blueprintEntry.Description,
-			ClientId:         &clientId,
+	if len(imageTypes) == 0 {
+		for _, imageRequest := range blueprint.ImageRequests {
+			composeRequest := ComposeRequest{
+				Customizations:   &blueprint.Customizations,
+				Distribution:     blueprint.Distribution,
+				ImageRequests:    []ImageRequest{imageRequest},
+				ImageName:        &blueprintEntry.Name,
+				ImageDescription: &blueprintEntry.Description,
+				ClientId:         &clientId,
+			}
+
+			composesResponse, err := h.handleCommonCompose(ctx, composeRequest, &blueprintEntry.VersionId)
+			if err != nil {
+				return err
+			}
+			composeResponses = append(composeResponses, composesResponse)
 		}
-		composesResponse, err := h.handleCommonCompose(ctx, composeRequest, &blueprintEntry.VersionId)
-		if err != nil {
-			return err
+	} else {
+		for _, imageRequest := range blueprint.ImageRequests {
+			for _, imageType := range imageTypes {
```

Comment:
I don't think this is correct, the nested loop will not do what we want.
What happens when I do have Blueprint with AWS, Azure, Bare metal and vmware targets and I ask for AWS, Azure and Bare metal images here? :thinking:

Reply:
Yes, this can be vastly improved.

We had an offline chat about how to approach this. Unfortunately, the [experimental package](https://pkg.go.dev/golang.org/x/exp/slices#Contains) `slices.Contains` cannot be used because [the type is generated from OpenAPI](https://github.com/osbuild/image-builder/blob/07736c30ffa4db61c5a9921e2e9584cc67d4da9d/internal/v1/api.go#L717) and it needs an annotation

To simplify the code, I suggest to create a helper function that will implement `containsImageType(it ImageType, its []ImageType)` on the particular type. Then use it in the loop, as I said above:

* when image type is `nil` this should mean "build me all types"
* when image type is empty slice this should mean "build nothing", in other words does not need special care